src/screens/Refunds/RefundUtils.res

```diff
@@ -137,20 +137,32 @@ let initialFixedFilter = () => [
   ),
 ]
 
+let getLabelFromFilterType = (filter: filter) => {
+  (filter :> string)
+}
+
+let getValueFromFilterType = (filter: filter) => {
+  switch filter {
+  | #connector_label => "merchant_connector_id"
+  | #status => "refund_status"
+  | _ => (filter :> string)
+  }
+}
+
 let getConditionalFilter = (key, dict, filterValues) => {
   open LogicUtils
 
   let filtersArr = switch key->getFilterTypeFromString {
-  | #connector_label => {
-      let arr = filterValues->getArrayFromDict("connector", [])->getStrArrayFromJsonArray
-      let newArr = arr->Array.flatMap(connector => {
-        let connectorLabelArr = dict->getDictfromDict("connector")->getArrayFromDict(connector, [])
-        connectorLabelArr->Array.map(item => {
-          item->getDictFromJsonObject->getString("connector_label", "")
-        })
+  | #connector_label =>
+    filterValues
```

Comment:
` filterValues
    ->getArrayFromDict("connector", [])
    ->getStrArrayFromJsonArray
    ->Array.flatMap(connector => {
      let connectorLabelArr = dict->getDictfromDict("connector")->getArrayFromDict(connector, [])
      connectorLabelArr->Array.map(item => {
        item->getDictFromJsonObject->getString("connector_label", "")
      })
    })`
    
    this part is reapeated numtiple times with diff keys, can it be made as a function which accepts a key and gives a array

Reply:
here, we are extracting connector from filter values and then corresponding labels from dict, it involves two diff json object, so clubing it will not help much i think